doc/source/releases.rst

```diff
@@ -0,0 +1,10 @@
+# Cutting a new release
+
+1. Update the version number in `version.py` using semver versioning rules.
+2. Tag your commit with the new version number. ie `git tag -a v0.1.0.dev0 -m "v0.1.0.dev0"` or `git tag -a v0.1.0 -m "v0.1.0" <commit hash>`
+3. Push the tag to GitHub. `git push origin v0.1.0`
+4. Install twine and build the package. `pip install twine` and `pip install build`
```

Comment:
[nit]: I'd suggest updating this to just say "install twine and build" since step 4 is just installing those two packages, and not building anything yet.